teps/0053-tekton-catalog-pipeline-organization.md

```diff
@@ -0,0 +1,477 @@
+   this](https://docs.google.com/document/d/1-czjvjfpuIqYKsfkvZ5RxIbtoFNLTEtOxaZB71Aehdg/edit#)
+
+3. When resources have parameters, resource or results, it must be
+   clear what values are allowed and the user must receive feedback
+   when they provide values that are not valid
+
+## Proposal
+
+### Support Tiers
+
+This proposal includes [three support
+levels](https://github.com/tektoncd/catalog/issues/5):
+
+1. **Community** - These are resources submitted by the community
+   which do not include [a
+   test](https://docs.google.com/document/d/1pZY7ROLuW47ymZYqUgAbxskmirrmDg2dd8VPATYXrxI/edit#heading=h.e9bh8qes6kh6) that can be tested successfully.
+
+   1. Not tested automatically
+
+   2. Not maintained by the tekton maintainers
+
+2. **Verified** - These are resources that include [a test](https://docs.google.com/document/d/1pZY7ROLuW47ymZYqUgAbxskmirrmDg2dd8VPATYXrxI/edit#heading=h.e9bh8qes6kh6)
+   that can be tested successfully but are owned by Tekton org members
+   outside of the OWNERS of the catalog.
+
+   3. Tested automatically
+
+   4. Not maintained by the tekton maintainers
+
+3. **Official** - These resources are verified but also are owned and
+   maintained by the Tekton catalog OWNERS, including being updated
+   [to be compatible with new versions of
+   Tekton.](https://docs.google.com/document/d/1pZY7ROLuW47ymZYqUgAbxskmirrmDg2dd8VPATYXrxI/edit#heading=h.t6fnduaxxkab)
+
+   5. Tested automatically
+
+   6. Maintained by the tekton maintainers
+
+   7. Image used are scanned periodically for CVE
+
+Those support tiers are mainly aimed for the Tekton Hub.
+
+### Versioning Resources
+
+Resources definition evolves across time. A new version of the image used is published, a new feature in tektoncd/pipeline is available, or a new feature / behaviour is added. On the other hand, the catalog should guarantee as much stability as possible for its user.
+
+A version is an identifiable information of a resource along with its kind (Task, Pipeline) and name, and it should be present in the definition of the resource. The location of a resource in the catalog should be computable given its Kind, Name and the Version. e.g. `/{kind}/{name}/{version}/{name}.yaml` - `pipeline/<pipeline-name>/0.5/<pipeline-name>.yaml`
+
+Adding a new version of a resource should be considered as a release of that resource. Here are some cases where a version bump for pipeline is necessary
+
+- Task used in the pipeline have new image
+
+  - For images that we own we should only bump the version if the new version breaks CI or if release notes include changes on the Tekton image
+  - For images that we do not own: it should be the responsibility of the author of the task to verify if changing an image version requires a new Task version
+  - When the image for a step is specified via an input params, we should treat changes to the default value of that param in the same way as changes to a step image
```

Comment:
im not sure that a new image in the Task is the important part here, the important thing is probably that the version of the Task was bumped (for some reason, which is probably out of scope for this TEP)

Reply:
So just adding, if the version of the task is bumped, simultaneously the the tasks used in the pipeline should be updated and the pipeline version should be bumped, would be fine right ?

Reply:
@PuneetPunamiya do we want it to be automatic? i would think the pipeline author would want to choose when to bump the version